commands/root.go

```diff
@@ -90,3 +107,28 @@ func Execute() {
 		os.Exit(1)
 	}
 }
+
+func getCommitAndDirty() (commit, dirty string) {
+	var d, c string
+
+	info, ok := debug.ReadBuildInfo()
+
+	if !ok {
+		fmt.Println("could not get commit")
+	}
+
+	// get the commit and
+	// modified status (that is the flag for repository dirty or not)
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs.revision":
```

Comment:
instead of going over all the values, you could simply try to read: `if val, ok := info.Settings["vcs.revision"]; ok { ... }`.

Reply:
`info.Settings` returns an array of structs, so I can't access required values by key name